docs/clusterctl.md

```diff
@@ -0,0 +1,87 @@
+
+## Overview
+
+`clusterctl` simplifies getting started with cluster-api-provider-aws. It wraps
+the complicated steps into one command that will bring you from nothing to a
+fully bootstrapped Kubernetes cluster running on AWS.
+
+## Install
+
+### From release
+
+Go to the
+[releases](https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases)
+page and pick a version. Each release contains several binaries one of which is
+`clusterctl`. Download the binary that was built for your architecture (either
+darwin (os x) or linux). Windows binaries are not supported at this time. Add
+the binary to your `$PATH` environment variable.
+
+### From source
+
+Clone the repo and run `make clusterctl`. Ensure the generated binary can be
+found in the `$PATH` environment variable. The binary is built with bazel and
+its location is `bazel-bin/cmd/clusterctl/darwin_amd64_pure_stripped/clusterctl`
+on a darwin system.
+
+## In depth
+
+### `clusterctl create`
+
+#### Bootstrap cluster
+
+The result of a successful `create` command is a cluster running in AWS with the
+cluster-api-provider-aws controllers watching Machine and Cluster objects ready
+to create as many clusters as you need.
+
+Cluster API controllers require kubernetes as a platform to
+run<sup>[1](#footnote1)</sup>. The first step to `clusterctl` is to create a
+bootstrap cluster. [Minikube](https://github.com/kubernetes/minikube) is used if
+no existing cluster exists. The bootstrap cluster's implementation is
+unimportant and any other software that creates a kubernetes cluster is
+acceptable. For example, it is possible to replace the Minikube usage with
+something like [Kubernetes-In-Docker](https://github.com/kubernetes-sigs/kind).
+However, you can provide a kubeconfig to `clusterctl` if you already have a
+cluster you'd like to use as a bootstrap cluster. Please see `clusterctl`'s help
+flag `--help`. Note: The bootstrap cluster *must* be >= v1.12.0 of Kubernetes.
+
+<a name="footnote1">1</a>: The controllers don't absolutely require a Kubernetes
+cluster to run on, but as they were designed to run on Kubernetes it is simply
+easier to create a cluster. There are plans to remove this dependency. Please
+see [this Request For
+Enhancement](https://github.com/kubernetes-sigs/cluster-api/issues/557).
+
+#### Cluster API Manager
+
+After the bootstrap cluster becomes active, `clusterctl` will apply the manager
+yaml. This will get the cluster-api-provider-aws' controllers running on the
+bootstrap cluster. These controllers watch the Machine and Cluster
+[CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+and react by interacting with the AWS API to create a Kubernetes cluster.
+
+#### Target cluster creation
+
+`clusterctl` will apply the generated cluster YAML after the bootstrap
+Kubernetes API server is ready to handle the CRDs. `clusterctl` lists the
+Machine and Cluster objects to determine if the bootstrap cluster is ready for
```

Comment:
It's not clear if the `clusterctl` list here is referring to waiting for the cluster/machine objects to be available after the provider components deployment (which should probably be mentioned before the Target cluster creation heading), or if it is for watching for the cluster object before creating the machine object.